synapse/notifier.py

```diff
@@ -315,6 +315,32 @@ async def on_new_room_events(
             event_entries.append((entry, event.event_id))
         await self.notify_new_room_events(event_entries, max_room_stream_token)
 
+    async def on_un_partial_stated_room(
+        self,
+        room_id: str,
+        new_token: int,
+    ) -> None:
+        """Used by the resync background processes to wake up all listeners
+        of this room that it just got un-partial-stated.
+
+        It will also notify replication listeners of the change in stream.
+        """
+
+        # Wake up all related user stream notifiers
+        user_streams = self.room_to_user_streams.get(room_id, set())
+        time_now_ms = self.clock.time_msec()
+        for user_stream in user_streams:
+            try:
+                user_stream.notify(
+                    StreamKeyType.UN_PARTIAL_STATED_ROOMS, new_token, time_now_ms
+                )
+            except Exception:
+                logger.exception("Failed to notify listener")
+
+        # Poke the replication so that other workers also see the write to
+        # the un-partial-stated rooms stream.
+        self.notify_replication()
```

Comment:
Is the point that we haven't changed the workers to _respond_ to the replication message we're putting out here?

I merged a few PRs last night and there were some conflicts---this might have been resolved elsewhere?

Reply:
Yes, sync workers do not wake up syncs when receiving the UnPartialStatedRoomStreamRow message.

This hasn't been resolved in any of in-flight PRs I've seen.

Reply:
It's also not critical to make it into the RC, since syncs should time out after 30 seconds and the un-partial stated room in question would appear in the next sync (?).

Reply:
Thanks, I've made this change in 995a2eff37